### DISCUSSION TITLE
collection().subscribe() for specific actions only

### DISCUSSION THREAD

**Discussion by RepComm** [pocketbase/pocketbase#2398] [Q&A]
Hi,
I am creating an unlimited grid of text, modelled as 'chunks'

chunks is a collection { x, y, text }

Every auth'd client can update the 'text' field by typing

each chunk is lazy-created as populated with text

I want the client to subscribe to the create action of chunks
It seems I can only subscribe wildcard to the chunks collection
or subscribe to a specific record

however, the client does not know the record ID yet as it isn't created

Instead, I can subscribe to the collection, but this (should?) listen to updates for all actions as well including update, streaming the entire chunk for each edit.

I am only interested in the creation of the chunks.

Is there I way I can subscribe to the specific action 'create' so that I don't waste data?

**Comment by RepComm** [comment]
Of course after I ask the question, the default behavior seems to be that it listens to creation by default and doesn't fire for updates.

```ts
db.ctx.collection("chunks").subscribe("*", (data)=>{
    console.log(data, data.action);
  });
  //{action: 'create', record: {…}} 'create'
  //{action: 'create', record: {…}} 'create'
 //etc
```
update doesn't seem to happen, and this is fine w/ my use case as I subscribe to that in a separate step.

Is this intended behavior? Am I using this right?

**Reply by ganigeorgiev** [reply]
I'm not sure that I understand what do you mean by _"listens to creation by default and doesn't fire for updates"_.

When subscribed to the wildcard topic (`*`) you'll receive a realtime event if the request client has List API rule access.

When subscribed to a single record id, you'll receive a realtime event if the request client has View API rule access.

---

> Instead, I can subscribe to the collection, but this (should?) listen to updates for all actions as well including update, streaming the entire chunk for each edit.
> I am only interested in the creation of the chunks.

You can filter the "create" events client-side in the callback by discarding every `data.action != 'create'`.

Alternatively, you can also discard the message server-side by attaching to the [`OnRealtimeBeforeMessageSend` hook](https://pocketbase.io/docs/event-hooks/#realtime-api-hooks) and return `hook.StopPropagation` for the update and delete actions if you don't want to stream messages for those.
